app/controllers/homepage_controller.rb

```diff
@@ -181,26 +181,72 @@ def find_listings(params, current_page, listings_per_page, filter_params, includ
 
-    ListingIndexService::API::Api.listings.search(
-      community_id: @current_community.id,
-      search: search,
-      includes: includes,
-      engine: FeatureFlagHelper.search_engine,
-      raise_errors: raise_errors
-      ).and_then { |res|
-      Result::Success.new(
-        if FeatureFlagHelper.search_engine == :discovery
-          res
-        else
+    if FeatureFlagHelper.feature_enabled?(:searchpage_v1)
+      DiscoveryClient.get(:query_listings,
+                          params: format_params(search.merge(marketplace_id: @current_community.id)))
+      .rescue {
+        Result::Error.new(nil, code: :discovery_api_error)
+      }
+        .and_then{ |res|
+        Result::Success.new(res[:body])
+      }
+    else
+      ListingIndexService::API::Api.listings.search(
+        community_id: @current_community.id,
+        search: search,
+        includes: includes,
+        engine: FeatureFlagHelper.search_engine,
+        raise_errors: raise_errors
+        ).and_then { |res|
+        Result::Success.new(
           ListingIndexViewUtils.to_struct(
             result: res,
             includes: includes,
             page: search[:page],
             per_page: search[:per_page]
           )
+        )
+      }
+    end
+  end
+
+  def format_params(original)
+    location_params =
+      if(original[:latitude].present? && original[:longitude].present?)
+        { :'search[lat]' => original[:latitude],
+          :'search[lng]' => original[:longitude],
+          :'search[distance_unit]' => original[:distance_unit],
+          :'search[scale]' => original[:scale],
+          :'search[offset]' => original[:offset],
+          :'filter[distance_max]' => original[:distance_max] }
+      else
+        {}
+      end
+
+    custom_fields = Maybe(original[:fields]).map { |fields|
+      fields.select { |f| [:numeric_range, :selection_group].include?(f[:type]) }
+      fields.map { |f|
+        if f[:type]  == :numeric_range
+          [:"custom[#{f[:id]}]", "double:#{f[:value].first}:#{f[:value].last}"]
+        else
+          [:"custom[#{f[:id]}]", "opt:#{f[:operator]}:#{f[:value].join(",")}"]
         end
-      )
-    }
+      }.to_h
+    }.or_else({})
+
+    {
+     :marketplace_id => original[:marketplace_id],
+     :'search[keywords]' => original[:keywords],
+     :'page[number]' => original[:page],
+     :'page[size]' => original[:per_page],
+     :'filter[price_min]' => Maybe(original[:price_cents]).map{ |p| p.min }.or_else(nil),
+     :'filter[price_max]' => Maybe(original[:price_cents]).map{ |p| p.max }.or_else(nil),
+     :'filter[omit_closed]' => !original[:include_closed],
+     :'filter[listing_shape_ids]' => Maybe(original[:listing_shape_ids]).join(",").or_else(nil),
+     :'filter[category_ids]' => Maybe(original[:categories]).join(",").or_else(nil),
+     :'search[locale]' => original[:locale],
+     :sort => original[:sort]
+    }.merge(location_params).merge(custom_fields).compact
```

Comment:
I'm not sure if `HomepageController` should be further bloated by Discovery-specific parameter formatting